packages/cli/src/UserManagement/decorators.ts

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/naming-convention */
+/* eslint-disable import/no-cycle */
+import { Application, Request, RequestHandler as Middleware, Response, Router } from 'express';
+import { posix } from 'path';
+import { ResponseHelper } from '..';
+import config from '../../config';
+
+interface RequestHandler<ReqBody, ResBody> extends Function {
+	async(req: Request<unknown, ResBody, ReqBody>, res: Response<ResBody>): Promise<ReqBody>;
+}
+
+export interface IRouter {
+	method: Method;
+	path: string;
+	handlerName: string | symbol;
+}
+
+const Keys = {
+	ROUTES: Symbol('routes'),
+	BASE_PATH: Symbol('base_path'),
+} as const;
+
+export enum Method {
+	GET = 'get',
+	POST = 'post',
+	PATCH = 'patch',
+	DELETE = 'delete',
+}
+
+export const RestController =
+	(basePath = '/'): ClassDecorator =>
+	(target: object) => {
+		Reflect.defineMetadata(Keys.BASE_PATH, basePath, target);
+	};
+
+const RouteFactory =
+	(...methods: Method[]) =>
+	(path = '/'): MethodDecorator =>
+	(target, handlerName) => {
+		const ControllerClass = target.constructor;
+		const routers: IRouter[] = Reflect.hasMetadata(Keys.ROUTES, ControllerClass)
+			? (Reflect.getMetadata(Keys.ROUTES, ControllerClass) as IRouter[])
+			: [];
```

Comment:
```suggestion
const routes: MetadataRoute[] = Reflect.getMetadata(Keys.ROUTES, ControllerClass) ?? [];
```

Reply:
i forgot that we wanted to start using null coalescing everywhere. thanks for catching this :+1: